Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Slate/FollowMeToggle.cs

```diff
@@ -1,40 +1,183 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using Microsoft.MixedReality.Toolkit.Utilities;
 using Microsoft.MixedReality.Toolkit.Utilities.Solvers;
+using System.Collections;
 using UnityEngine;
 
 namespace Microsoft.MixedReality.Toolkit.UI
 {
     [RequireComponent(typeof(RadialView))]
     public class FollowMeToggle : MonoBehaviour
     {
+        /// <summary>
+        /// An optional object for visualizing the 'Follow Me' mode state.
+        /// </summary>
+        public GameObject VisualizationObject
+        {
+            get { return visualizationObject; }
+            set { visualizationObject = value; }
+        }
+
         [SerializeField]
-        [Tooltip("An opional object for visualizing the 'Follow Me' mode state")]
+        [Tooltip("An optional object for visualizing the 'Follow Me' mode state.")]
         private GameObject visualizationObject = null;
 
+        /// <summary>
+        /// An optional Interactable to select/deselect when toggling the follow behavior.
+        /// </summary>
+        public Interactable InteractableObject
+        {
+            get { return interactableObject; }
+            set { interactableObject = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("An optional Interactable to select/deselect when toggling the follow behavior.")]
+        private Interactable interactableObject = null;
+
+        /// <summary>
+        /// Should following be automatically enabled when the user is further than a certain distance away?
+        /// </summary>
+        public bool AutoFollowAtDistance
+        {
+            get { return autoFollowAtDistance; }
+            set
+            {
+                if (autoFollowAtDistance != value)
+                {
+                    autoFollowAtDistance = value;
+
+                    if (autoFollowAtDistance)
+                    {
+                        autoFollowDistanceCheck = StartCoroutine(AutoFollowDistanceCheck());
+                    }
+                    else
+                    {
+                        StopCoroutine(autoFollowDistanceCheck);
+                    }
+                }
+            }
+        }
+
+        [SerializeField]
+        [Tooltip("Should following be automatically enabled when the user is further than a certain distance away?")]
+        private bool autoFollowAtDistance = false;
```

Comment:
What happens if somebody checks autoFollowAtDistance in the editor during runtime?

Reply:
I suppose if they did that nothing would happen (which might not be desirable?). If they checked that option at runtime in the inspector and _then_ called the settor that could cause a routine leak. Maybe we should add an OnValidate which calls the settor to support that case.

Reply:
>I suppose if they did that nothing would happen (which might not be desirable?).

Yes, can we make it work? Especially since this is not experimental, let's get it right.

Reply:
>If they checked that option at runtime in the inspector and then called the settor that could cause a routine leak. Maybe we should add an OnValidate which calls the settor to support that case.

Good idea

Reply:
Of course! That sounds good to me. @cre8ivepark please let me know if you need assistance making these changes.

Reply:
@Cameron-Micka Yes, please!

Reply:
Just pushed up a change that makes the FollowMeToggle resilient to runtime inspector changes.